src/v/cluster/topics_frontend.cc

```diff
@@ -1318,4 +1321,92 @@ topics_frontend::generate_reassignments(
       });
 }
 
+namespace {
```

Comment:
Generally the calls in the topic table are synchronized with their respective controller commands. i.e `create_topic` calls `replicate` via the `create_topic_cmd` and so on and so forth with the rest. 

Notice there's no `list_offsets` or `list_topic` method in the `topics_frontend` API. That is because there's no controller command for those, because they don't mutate controller state, in order words, we don't push a command to raft0 on invocation of `list_topics`.

Your support for `list_partition_reassignments` follows this same paradigm. No controller state is mutated, therefore we should drop this commit and just move the logic into the `list_partition_reassignments` handler.